### PR TITLE
fix(upgrade): component injectors should not link the module injector…

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: packages/upgrade/src/static/upgrade_module.ts [hunk 1]
```diff
@@ -234,3 +238,15 @@ export class UpgradeModule {
...
+
+class NgAdapterInjector implements Injector {
+  constructor(private injector: Injector) {}
+
+  get(token: any, notFoundValue?: any): any {
+    if (notFoundValue === NOT_FOUND_CHECK_ONLY_ELEMENT_INJECTOR) {
+      return notFoundValue;
+    }
+
+    return this.injector.get(token);
```
Comment (gkalpak) [inline packages/upgrade/src/static/upgrade_module.ts]: Pass `notFoundValue` here too?

Comment (vicb) [inline packages/upgrade/src/static/upgrade_module.ts]: yep, thanks !

File: packages/upgrade/test/static/integration/downgrade_component_spec.ts [hunk 2]
```diff
@@ -361,5 +361,59 @@ export function main() {
            expect(multiTrim(document.body.textContent)).toBe('parent(child)');
          });
        }));
+
+    it('should work with ng2 lazy loaded components=', async(() => {
+
+         let componentInjector: Injector;
+
+         @Component({selector: 'ng2', template: '[<ng-content></ng-content>]'})
```
Comment (gkalpak) [inline packages/upgrade/test/static/integration/downgrade_component_spec.ts]: Nit: No need for `<ng-content>` (afaict). You can just have `'ng2'` (or an empty template if you don't care about it).

File: packages/upgrade/test/static/integration/downgrade_component_spec.ts [hunk 3]
```diff
@@ -361,5 +361,59 @@ export function main() {
...
+
+         @NgModule({
+           declarations: [LazyLoadedComponent],
+           entryComponents: [LazyLoadedComponent],
+         })
+         class LazyLoadedModule {
+         }
+
+         const ng1Module = angular.module('ng1', [])
+                               .directive('ng1', () => ({template: '<ng2></ng2>'}))
```
Comment (gkalpak) [inline packages/upgrade/test/static/integration/downgrade_component_spec.ts]: Nit: No need for this directive. You can use `const element = html('<ng2></ng2>')` below instead.

File: packages/upgrade/test/static/integration/downgrade_component_spec.ts [hunk 4]
```diff
@@ -361,5 +361,59 @@ export function main() {
...
+                               .directive('ng2', downgradeComponent({component: Ng2Component}));
+
+         const element = html('<ng1></ng1>');
+
+         platformBrowserDynamic().bootstrapModule(Ng2Module).then((modRef) => {
```
Comment (gkalpak) [inline packages/upgrade/test/static/integration/downgrade_component_spec.ts]: Nit: You could probably use the bootstrap helper (and save two whole lines of code 😛):

```ts
bootstrap(platformBrowserDynamic(), Ng2Module, element, ng1Module).then(upgrade => {
  const modInjector = upgrade.injector;
  // Emulate the router lazy loading ...
  ...
});
```

**Review by gkalpak — APPROVED** [review]
LGTM (just a couple of nits)

File: packages/upgrade/src/static/upgrade_module.ts [hunk 5]
```diff
@@ -234,3 +238,19 @@ export class UpgradeModule {
     }
   }
 }
+
+class NgAdapterInjector implements Injector {
+  constructor(private modInjector: Injector) {}
+
+  // When Angular locate a service in the component injector tree, the not found value is set to
+  // `NOT_FOUND_CHECK_ONLY_ELEMENT_INJECTOR` in such a case we should not walk up to the module
```
Comment (tbosch) [inline packages/upgrade/src/static/upgrade_module.ts]: Add a `.` before `in such case`

File: packages/upgrade/test/static/integration/downgrade_component_spec.ts [hunk 6]
```diff
@@ -361,5 +361,59 @@ export function main() {
...
+           const modInjector = modRef.injector;
+
+           const adapter: UpgradeModule = modInjector.get(UpgradeModule);
+           adapter.bootstrap(element, [ng1Module.name]);
+
+           // Emulate the router lazy loading a module and creating a component
+           const compiler = modInjector.get(Compiler);
+           const modFactory = compiler.compileModuleSync(LazyLoadedModule);
+           const childMod = modFactory.create(modInjector);
+           const cmpFactory = childMod.injector.get(ComponentFactoryResolver)
```
Comment (tbosch) [inline packages/upgrade/test/static/integration/downgrade_component_spec.ts]: childMod.componentFactoryResolver

**Review by tbosch — APPROVED** [review]

File: packages/upgrade/test/static/integration/downgrade_component_spec.ts (hunk 2 above)
Comment (vicb) [inline packages/upgrade/test/static/integration/downgrade_component_spec.ts]: right, I tried many things before coming to this test... will remove

File: packages/upgrade/test/static/integration/downgrade_component_spec.ts [hunk 7]
```diff
@@ -361,5 +361,55 @@ export function main() {
            expect(multiTrim(document.body.textContent)).toBe('parent(child)');
          });
        }));
+
+    it('should work with ng2 lazy loaded components=', async(() => {
```
Comment (gkalpak) [inline packages/upgrade/test/static/integration/downgrade_component_spec.ts]: Nit: _components=_ --> _components_

**Pull request merged**